### PR TITLE
[SPARK-31117][SQL][TEST] reduce the test time of DateTimeUtilsSuite

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

`DateTimeUtilsSuite.daysToMicros and microsToDays` takes 30 seconds, which is too long for a UT.

This PR changes the test to check random data, to reduce testing time. Now this test takes 1 second.

### Why are the changes needed?

make test faster

### Does this PR introduce any user-facing change?

no

### How was this patch tested?

N/A

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @MaxGekk @HyukjinKwon

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/DateTimeUtilsSuite.scala [hunk 1]
```diff
@@ -623,10 +623,12 @@ class DateTimeUtilsSuite extends SparkFunSuite with Matchers with SQLHelper {
       "MIT" -> Set(15338))
     for (tz <- ALL_TIMEZONES) {
       val skipped = skipped_days.getOrElse(tz.getID, Set.empty)
-      (-20000 to 20000).foreach { d =>
+      val testingData = Seq(-20000, 20000) ++
```
Comment (cloud-fan) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/DateTimeUtilsSuite.scala]: I don't know why we pick -20000 and 20000 as the boundaries, just to be safe to always test the boundary values.

**Review by HyukjinKwon — APPROVED** [review]

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/DateTimeUtilsSuite.scala [hunk 2]
```diff
@@ -623,10 +623,12 @@ class DateTimeUtilsSuite extends SparkFunSuite with Matchers with SQLHelper {
...
+        (1 to 1000).map(_ => (math.random() * 40000 - 20000).toInt)
```
Comment (MaxGekk) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/DateTimeUtilsSuite.scala]: There are 2 approaches for generating random values for Catalyst's types:
1. https://github.com/apache/spark/blob/1febd373ea806326d269a60048ee52543a76c918/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/LiteralGenerator.scala#L114-L124
2. RandomDataGenerator.forType(): https://github.com/apache/spark/blob/9562b26914f9a0c513e34b9b90cdec60067ef055/sql/catalyst/src/test/scala/org/apache/spark/sql/RandomDataGenerator.scala#L178-L193

Can you use one of them?

**Comment by cloud-fan** [comment]
thanks, merging to master/3.0!

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/DateTimeUtilsSuite.scala (hunk 2 above)
Comment (cloud-fan) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/DateTimeUtilsSuite.scala]: Sorry seeing this comment too late.

Here we just need to get a random int. It's nothing about the catalyst data type. I think it's better to keep simple.

Comment (MaxGekk) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/DateTimeUtilsSuite.scala]: It is not related to the PR but I think we could unify random data generators for Catalyst's types.

**Pull request closed without merging**